### ISSUE TITLE
Colon seperated namespace doesn't create properly

### ISSUE THREAD

**Issue by Axemasta** [dansiegel/Prism-Templates#47]
# Description

My companies house style is to name projects like so: Company.Projectname.ProjectType, This means I am creating a project called "mycompany.myapp.app" When this creates, the namespaces are not correctly implemented:

This should created the namespace `mycompany.myapp.app`, however in the iOS / droid project, it will create `mycompany_myapp.app` as the namespace. This means a blank project won't build & I can't seem to rename everything to correct this issue.

## Version Info

```

dotnet --info
.NET Command Line Tools (2.1.4)

Product Information:
 Version:            2.1.4
 Commit SHA-1 hash:  5e8add2190

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.13
 OS Platform: Darwin
 RID:         osx.10.12-x64
 Base Path:   /usr/local/share/dotnet/sdk/2.1.4/

Microsoft .NET Core Shared Framework Host

  Version  : 2.0.5
  Build    : 17373eb129b3b05aa18ece963f8795d65ef8ea54
```

## Project Command

I installed prism templates for VS mac, and created a new solution.

**Comment by dansiegel** [comment]
If you're creating the template from the VSMac extension I specifically test `Contoso.AwesomeApp` like the placeholder shows... Ultimately if it's throwing an underscore in the name the problem is with the dotnet template engine shipped in VSMac but I'll see if I can replicate and what I can do to fix it... there's already a few template engine & VSMac bugs that I'm fixing (hence the delay in the project creation)

**Comment by Axemasta** [comment]
@dansiegel as a workaround I was eventually able to get my namespace name how It needed to be. I ended up making an app with a namespace that didn't have the dots (everything else did) and then renaming the namespace in the files & project. Everything now works nicely, I guess its probably a time consuming fix for an issue that won't be all that common!

**Comment by dansiegel** [comment]
I've purposefully made sure that a naming schema like `Company.ProductName` is supported in the templates, and `Company.ProductName.ModuleName` is supported in the Module Template. I just want to be sure your issue is that you have something like `Company.Product.Scope` for the project name though right?

**Comment by Axemasta** [comment]
Yes my naming scheme was `Company.Product.Scope`, to get around the issue i named my app `CompanyProductScope` and then replaced the namespaces with  `Company.Product.Scope` (as I had tested this in a non prism project)

**Comment by dansiegel** [comment]
thanks for letting me know about the issue I'll get a work around in as soon as I can
